### PR TITLE
Diff configuration in syscheck page

### LINKED ISSUE #2554: Add new report_changes options to FIM configuration page
## Description
With wazuh/wazuh#5157, several new features have been added to the report_changes option in FIM. These features need to be added to the FIM configuration page.

## Example of configuration
```xml
<directories report_changes="yes" diff_size_limit="50MB">/home/user</directories>

<diff>
    <disk_quota>
        <enabled>yes</enabled>
        <limit>50KB</limit>
    </disk_quota>
    <file_size>
        <enabled>yes</enabled>
        <limit>5KB</limit>
    </file_size>
    <nodiff>/home/user/file</nodiff>
</diff>
```

### PR DESCRIPTION
**Pull request by jesusjimsa**
## Description

This pull request adds the configuration of the new diff section (wazuh/wazuh#5157) to the syscheck configuration page.

```xml
<directories report_changes="yes" diff_size_limit="50MB">/home/user</directories>

<diff>
    <disk_quota>
        <enabled>yes</enabled>
        <limit>1GB</limit>
    </disk_quota>
    <file_size>
        <enabled>yes</enabled>
        <limit>50MB</limit>
    </file_size>

    <nodiff>/etc/ssl/private.key</nodiff>
</diff>
```

This pull request closes #2554.

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [x] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

### CODE REVIEW

File: source/user-manual/reference/ossec-conf/syscheck.rst [hunk 1]
```diff
@@ -834,6 +825,110 @@ Set the maximum synchronization message throughput.
 | **Allowed values** | Integer number between 0 and 1000000. 0 means disabled. |
 +--------------------+---------------------------------------------------------+
 
+.. _reference_ossec_syscheck_diff:
+
+diff
+^^^^
+
+.. versionadded:: 3.14
```
Comment (bah07) [inline source/user-manual/reference/ossec-conf/syscheck.rst:833]: Please, change the version to 4.0

File: source/user-manual/reference/ossec-conf/syscheck.rst [hunk 2]
```diff
@@ -834,6 +825,110 @@ Set the maximum synchronization message throughput.
...
+
+      <nodiff>/etc/ssl/private.key</nodiff>
+    </diff>
+
+disk_quota
+""""""""""
+
+.. versionadded:: 3.14
+
+This option can be used to limit the size of the ``queue/diff/local`` folder where Wazuh stores the compressed files used to perform the diff operation when ``report_changes`` is enabled. After reaching this size, alerts will not show the diff information until the size is smaller than the configured limit again.
```
Comment (bah07) [inline source/user-manual/reference/ossec-conf/syscheck.rst:857]: Please, remove again word at the end.

File: source/user-manual/reference/ossec-conf/syscheck.rst (hunk 1 above)
Comment (jesusjimsa) [reply source/user-manual/reference/ossec-conf/syscheck.rst]: Done 3e89f63

File: source/user-manual/reference/ossec-conf/syscheck.rst (hunk 2 above)
Comment (jesusjimsa) [reply source/user-manual/reference/ossec-conf/syscheck.rst]: Done 3e89f63

File: source/user-manual/reference/ossec-conf/syscheck.rst [hunk 3]
```diff
@@ -834,6 +825,110 @@ Set the maximum synchronization message throughput.
...
+
+.. versionadded:: 4.0
+
+This option can be used to limit the size of the ``queue/diff/local`` folder where Wazuh stores the compressed files used to perform the diff operation when ``report_changes`` is enabled. After reaching this size, alerts will not show the diff information until the size is smaller than the configured limit.
+
+**enabled**
+
+.. versionadded:: 4.0
+
+Specifies whether there will be a disk quota limit or not.
```
Comment (bah07) [inline source/user-manual/reference/ossec-conf/syscheck.rst:863]: Set the quota limit option to enable or disable.

File: source/user-manual/reference/ossec-conf/syscheck.rst [hunk 4]
```diff
@@ -834,6 +825,110 @@ Set the maximum synchronization message throughput.
...
+
+.. versionadded:: 4.0
+
+This option can be used to limit the size of the file which will report diff information with ``report_changes`` enabled. Files bigger than this limit will not report diff information until the size is smaller than the configured limit again.
+
+**enabled**
+
+.. versionadded:: 4.0
+
+Specifies whether there will be a file size limit or not.
```
Comment (bah07) [inline source/user-manual/reference/ossec-conf/syscheck.rst:894]: Set the size limit of a file to enable or disable.

File: source/user-manual/reference/ossec-conf/syscheck.rst [hunk 5]
```diff
@@ -89,6 +89,16 @@ Attributes:
 +                          +------------------------------------------------------------+----------------------------------------------------------+
 |                          | Allowed values                                             | yes, no                                                  |
 +--------------------------+------------------------------------------------------------+----------------------------------------------------------+
+| **diff_size_limit**      | Limit the mazximum size of the file which will report diff information with ``report_changes`` enabled.               |
++                          +                                                                                                                       +
+|                          | Files bigger than this value will not report diff information.                                                        |
++                          +                                                                                                                       +
+|                          | .. versionadded:: 4.0.0                                                                                              |
```
Comment (bah07) [inline source/user-manual/reference/ossec-conf/syscheck.rst:96]: Space is missing that causes a warning when generating the table.

**Review by bah07 — CHANGES_REQUESTED** [review]

File: source/user-manual/reference/ossec-conf/syscheck.rst [hunk 6]
```diff
@@ -89,6 +89,16 @@ Attributes:
 +                          +------------------------------------------------------------+----------------------------------------------------------+
 |                          | Allowed values                                             | yes, no                                                  |
 +--------------------------+------------------------------------------------------------+----------------------------------------------------------+
+| **diff_size_limit**      | Limit the mazximum size of the file which will report diff information with ``report_changes`` enabled.               |
```
[the remark this answers is not in the document]
Comment (jesusjimsa) [reply source/user-manual/reference/ossec-conf/syscheck.rst]: Done 8cce8d0

File: source/user-manual/reference/ossec-conf/syscheck.rst (hunk 4 above)
Comment (jesusjimsa) [reply source/user-manual/reference/ossec-conf/syscheck.rst]: Done 8cce8d0

File: source/user-manual/reference/ossec-conf/syscheck.rst (hunk 3 above)
Comment (jesusjimsa) [reply source/user-manual/reference/ossec-conf/syscheck.rst]: Done 8cce8d0

File: source/user-manual/reference/ossec-conf/syscheck.rst (hunk 5 above)
Comment (jesusjimsa) [reply source/user-manual/reference/ossec-conf/syscheck.rst]: Done 8cce8d0

**Review by davidfmiranda — CHANGES_REQUESTED** [review]
Hello @jesusjimsa. LGTM, nevertheless, there are conflicts to be solved before merging

**Review by bah07 — APPROVED** [review]

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Pull request merged**